src/index.js

```diff
@@ -175,6 +171,39 @@ async function check(autoCheckElement: AutoCheckElement) {
   }
 }
 
+async function processResponse(response: Response, input: HTMLInputElement, required: boolean) {
+  if (response.status === 200) {
+    if (required) {
+      input.setCustomValidity('')
+    }
+    input.dispatchEvent(
+      new CustomEvent('auto-check-success', {
+        bubbles: true,
+        detail: {
+          response: response.clone()
+        }
+      })
+    )
+    return
+  }
+
+  const messages = []
+  const setValidity = result => messages.push(result)
+  input.dispatchEvent(
+    new CustomEvent('auto-check-error', {
+      bubbles: true,
+      detail: {
+        response: response.clone(),
+        setValidity
+      }
+    })
+  )
+
+  if (!required) return
+  const message = messages.length ? await Promise.race(messages) : 'Input is not valid'
```

Comment:
Curious why this takes an array of promises instead of just one Promise that gets re-assigned? What would be a scenario where this instance of `setValidity` get called multiple times? 🤔 

We might've discussed this during `text-expander` pairings but I don't remember the specifics. 🙈